integration-tests/src/test/java/io/datanerds/grpc/ConsulNameResolverTest.java

```diff
@@ -0,0 +1,276 @@
+package io.datanerds.grpc;
+
+import com.google.common.net.HostAndPort;
+import com.orbitz.consul.*;
+import com.orbitz.consul.model.health.ServiceHealth;
+import com.pszymczyk.consul.junit.ConsulResource;
+import io.datanerds.grpc.ConsulQueryParameter.Builder;
+import io.datanerds.grpc.integrationtests.PingPongGrpc;
+import io.datanerds.grpc.integrationtests.PingPongProto;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.UUID;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+public class ConsulNameResolverTest {
+    private static final Logger logger = LoggerFactory.getLogger(ConsulNameResolverTest.class);
+
+    @ClassRule
+    public static final ConsulResource consulResource = new ConsulResource();
+
+    @Test
+    public void checkIfConsulIsUpAndHealthy() throws Throwable {
+        await().atMost(15, SECONDS).until(() -> {
+            try {
+                Consul consul = Consul.builder().withHostAndPort(HostAndPort.fromParts("localhost", consulResource
+                        .getHttpPort())).build();
+                HealthClient healthClient = consul.healthClient();
+                List<ServiceHealth> nodes = healthClient.getHealthyServiceInstances("consul").getResponse();
+                return !nodes.isEmpty();
+            } catch (ConsulException ex) {
+                logger.debug("Can't connect to Consul", ex);
+                return false;
+            }
+        });
+    }
+
+    @Test
+    public void checkPingPongServer() throws Exception {
+        int port = 55555;
+        PingPongServer server = new PingPongServer(port);
+        server.start();
+
+        ManagedChannel channel = ManagedChannelBuilder
+                .forTarget("localhost:" + port)
+                .usePlaintext(true)
+                .build();
+
+        PingPongGrpc.PingPongBlockingStub blockingStub = PingPongGrpc.newBlockingStub(channel);
+        PingPongProto.Ping ping = PingPongProto.Ping.newBuilder().setPing("PING").build();
+        PingPongProto.Pong pong = blockingStub.pingPong(ping);
+        assertEquals("PONG-" + port + ": " + ping.getPing(), pong.getPong());
```

Comment:
`assertThat(lala, is(equalTo(pong.getPong)))` will make failures and assertation more readable